### PR TITLE
feat(ci): add Docker Scout vulnerabilities scanning

### PR DESCRIPTION
**Pull request by gustavovalverde**
## Motivation

We must avoid publishing new releases without being fully aware of any new vulnerabilities that might be introduced into the image. This ensures we have visibility and can take the necessary actions, such as updating our READMEs, fixing the vulnerabilities, or implementing any other required measures.

### Specifications & References

- https://docs.docker.com/scout/explore/analysis/

## Solution

- Implement Docker Scout CLI with their GitHub Action: https://github.com/docker/scout-action
- As this action is too verbose in PRs for our use case, just limit it to the `Release` PR until we can have a better implementation with https://github.com/docker/scout-action/issues/56
- Record the results in Docker Scout in a `prod`, `stage` and `dev` environment for future reference https://docs.docker.com/reference/cli/docker/scout/environment/

### Tests

- I've triggered multiple changes in our Dockerfile, and in this PR title, to confirm the action's output (below) and behavior

### Follow-up Work

- Implementing policies for Docker Scout that are compliant with our supply chain requirements https://docs.docker.com/scout/policy/
- https://github.com/docker/scout-action/issues/56

### PR Author's Checklist

- [x] The PR name will make sense to users.
- [x] The PR provides a CHANGELOG summary.
- [x] The solution is tested.
- [x] The documentation is up to date.
- [x] The PR has a priority label.

### PR Reviewer's Checklist

- [ ] The PR Author's checklist is complete.
- [ ] The PR resolves the issue.

### CODE REVIEW

**Comment by gustavovalverde** [comment]
Need a workaround for: https://github.com/docker/scout-action/issues/16

File: .github/workflows/sub-build-docker-image.yml [hunk 1]
```diff
@@ -179,3 +181,32 @@ jobs:
...
+        if: ${{ inputs.dockerfile_target == 'runtime' && contains(github.event.pull_request.title, 'Release v') }}
+        with:
+          command: cves,recommendations,compare
+          image: us-docker.pkg.dev/${{ vars.GCP_PROJECT }}/zebra/${{ inputs.image_name }}:${{ steps.meta.outputs.version }}
+          to: zfnd/zebra:latest
+          ignore-base: true
+          ignore-unchanged: true
+          only-fixed: true
+          only-severities: critical,high
+          environment: ${{ (github.event_name == 'release' && !github.event.release.prerelease && 'prod') || (github.event_name == 'push' && github.ref_name == 'main' && 'stage') || (github.event_name == 'pull_request' && 'dev') }}
```
Comment (upbqdn) [inline .github/workflows/sub-build-docker-image.yml:210]: @gustavovalverde, could you link to the documentation for the `environment` input? According to https://github.com/docker/scout-action/blob/main/README.md, the `environment` input is associated only with the `environment` command which we don't use.

Comment (gustavovalverde) [reply .github/workflows/sub-build-docker-image.yml:210]: Oh, this is actually from the Docker Scout Action: https://docs.docker.com/reference/cli/docker/scout/environment/

File: .github/workflows/sub-build-docker-image.yml [hunk 2]
```diff
@@ -179,3 +181,32 @@ jobs:
...
+      - name: Docker Scout
+        id: docker-scout
+        uses: docker/scout-action@v1.13.0
+        # We only run Docker Scout on the `runtime` target, as the other targets are not meant to be released
+        # and are commonly used for testing, and thus are ephemeral.
+        # TODO: Remove the `contains` check once we have a better way to determine if just new vulnerabilities are present.
+        # See: https://github.com/docker/scout-action/issues/56
+        if: ${{ inputs.dockerfile_target == 'runtime' && contains(github.event.pull_request.title, 'Release v') }}
+        with:
+          command: cves,recommendations,compare
```
Comment (upbqdn) [inline .github/workflows/sub-build-docker-image.yml]:
```suggestion
          command: cves,recommendations,compare,environment
```

File: .github/workflows/sub-build-docker-image.yml (hunk 1 above)
Comment (upbqdn) [reply .github/workflows/sub-build-docker-image.yml:210]: Don't we then miss the `environment` command at line 203?

**Review by upbqdn — DISMISSED** [review]

File: .github/workflows/sub-build-docker-image.yml (hunk 1 above)
Comment (gustavovalverde) [reply .github/workflows/sub-build-docker-image.yml:210]: Yes, added your recommendation

**Review by upbqdn — APPROVED** [review]

**Pull request merged**